### ISSUE TITLE
Fields of extern structs produce name collisions

### ISSUE THREAD

**Issue by halfdan** [crystal-lang/crystal#8773]
**Include reproducible code**: we should be able to paste it into an editor, compile and run it and get the same error as you. Otherwise it's impossible for us to reproduce the bug.

```
@[Extern]
struct Test1
    reserved0 : LibC::Int
end

@[Extern]
struct Test2
    reserved0 : LibC::Int
end
```

Include Crystal compiler version (`crystal -v`) and OS. If possible, try to see if the bug still reproduces on master.

```
✗ crystal -v
Crystal 0.32.1 (2019-12-18)

LLVM: 9.0.0
Default target: x86_64-apple-macosx
```

**Comment by Sija** [comment]
```
Error: variable 'reserved0' already declared
```

https://carc.in/#/r/8j8g

**Comment by asterite** [comment]
I think this should be fixed (variables should be scoped to a class), but in any case the syntax for declaring instance variables of extern structs is not like that, you still need to use `@x : Int32` or `property`, etc.

So low priority to fix this.

**Comment by asterite** [comment]
Related to #4106 and #4407 (I'd say a duplicate of the latter)

**Comment by straight-shoota** [comment]
Yeah, this ha pretty much been reported in #4407 and is actually not specific to extern structs:
```cr
module Test1
  var : Int32
end

module Test2
  var : Int32 # Error: variable 'var' already declared
end
```

I'd still keep this as a dedicated issue because #4407 is actually two different issues.

**Comment by halfdan** [comment]
This does not seem to happen when the structs are declared inside a lib.

**Comment by asterite** [comment]
Yes, because that syntax inside structs inside a lib is for declaring fields, not local variables. That's why it doesn't break in that case (it's handled differently).

**Comment by Sija** [comment]
It might be difficult to immediately see the logic difference between:

```cr
struct Test
  reserved : LibC::Int
end
```

and

```cr
struct Test
  @reserved : LibC::Int
end
```

Both examples compile without a problem.

**Comment by RX14** [comment]
@halfdan the language which exists inside `lib` blocks as an entirely different language to Crystal, it's just a DSL. It's not the best solution but it's good enough and it works.
